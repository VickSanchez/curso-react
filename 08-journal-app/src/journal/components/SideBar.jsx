import { ListItemText, Box, Drawer, Toolbar, Typography, Divider, List, ListItem, ListItemButton, ListItemIcon, Grid } from '@mui/material';
import { TurnedInNot } from '@mui/icons-material';

export const SideBar = ({ drawerWidth }) => {
  return (
    
    <Box
        component='nav'
        sx={{ width: { sm: drawerWidth }, flexShrink: { sm: 0 } }}
    >
        <Drawer
            variant='permanent'
            open
            sx={{
                display: { xs: 'block'},
                '& .MuiDrawer-paper': { boxSizing: 'border-box', width: drawerWidth }
            }}
            >

                <Toolbar>
                    <Typography varinat='h6' noWrap component='div'>Vick Sanchez</Typography>
                </Toolbar>
                <Divider/>

                <List>  
                    {
                        [ 'Enero', 'Febrero', 'Marzo', 'Abril'].map( text => (
                            <ListItem key={ text } disablePadding >
                                <ListItemButton>
                                    <ListItemIcon>
                                        <TurnedInNot/>
                                    </ListItemIcon>
                                    <Grid container>
                                        <ListItemText primary={ text } />
                                        <ListItemText secondary={ 'Ullamco laboris aute pariatur enim veniam eiusmod ullamco Lorem ullamco.'}/>
                                    </Grid>
                                </ListItemButton>

                            </ListItem>
                        ))
                    }
                </List>


        </Drawer>


    </Box>
    
  )
}
