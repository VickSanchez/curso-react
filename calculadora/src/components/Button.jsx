import React, { useContext, memo, useCallback } from "react";
import { CalcContext } from "../context/CalcContext";
import "./styles/Button.css";

export const Button = React.memo(
  ({ symbol, color, handleClick, symbolSec, nombreClase }) => {
    const { addToText, resetInput, calculateResult, text } =
      useContext(CalcContext);

    const onClick = () => {
      handleClick === "calculateResult"
        ? calculateResult(text)
        : handleClick === "resetInput"
        ? resetInput()
        : addToText(symbol);
    };

    return (
      <div
        onClick={onClick}
        className={
          !!nombreClase ? `button-wrapper ${nombreClase}` : "button-wrapper"
        }
        style={{ backgroundColor: color }}
      >
        {!!symbolSec ? symbolSec : symbol}
      </div>
    );
  }
);
