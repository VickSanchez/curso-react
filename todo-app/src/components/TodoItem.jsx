import { Button } from "primereact/button"
import { Dialog } from "primereact/dialog";
import { useState } from "react";

export const TodoItem = ({ todo }) => {
    const [displayBasic, setDisplayBasic] = useState(false);
    const [position, setPosition] = useState('center');
    const renderFooter = (name) => {
        return (
            <div>
                <Button label="Tarea Completada" icon="pi pi-check" onClick={() => onHide(name)} autoFocus />
            </div>
        );
    }
    const onHide = (name) => {
        return dialogFuncMap[`${name}`](false);
    }
    const dialogFuncMap = {
        'displayBasic': setDisplayBasic,
    }
    const onClick = (name, position) => {
        dialogFuncMap[`${name}`](true);

        if (position) {
            setPosition(position);
        }
    }
    const getStyle = () => {
        return{
            textDecoration: todo.done ? 'line-through': 'none',
        }
    }
  return (
    <div className="todo">

    <Button label={todo.titulo} onClick={() => onClick('displayBasic')} className="btn-todo" style={getStyle()} />
    
    <Dialog header={todo.titulo} visible={displayBasic} style={{ width: '50vw' }} footer={renderFooter('displayBasic')} onHide={() => onHide('displayBasic')}>
    <p>{todo.descripcion}</p>
    </Dialog> 

    <Button icon="pi pi-times" onClick={() => onHide(name)} className="btn-borrar" />
   
    </div>
  )
}
