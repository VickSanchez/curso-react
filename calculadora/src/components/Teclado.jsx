import React from "react";
import { teclas } from "../helper/teclasData";
import { Button } from "./Button";

export const TecladoApp = () => {
    

  return (
    <>
      {teclas.map((tecla, i) => (
        
          setTimeout(() => {(
            <Button key={i} symbol={tecla.symbol} symbolSec={tecla.symbolSec} handleClick={tecla.handleClick} nombreClase={tecla.nombreClase}
            />)
            
          }, 100)
        
      ))}
    </>
  );
};
