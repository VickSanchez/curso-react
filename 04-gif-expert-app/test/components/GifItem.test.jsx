import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"

describe('first', () => { 

    const title = "titulo";
    const url = "www.hola.com";

    test('Debe de hacer match con el Snapshot', () => { 

        const { container } = render( <GifItem title={ title }url={ url }/>)
    
        expect( container ).toMatchSnapshot();
    });

    // test('Debe de mostrar la imagen con el URL y el ALT indicado', () => { 
        
    //     render( <GifItem title={ title }url={ url }/>)

    //     const { src, alt } = screen.getByRole( 'img' );
    //     expect( src ).toBe( url );
    //     expect( alt ).toBe( title );
    //  })
 });