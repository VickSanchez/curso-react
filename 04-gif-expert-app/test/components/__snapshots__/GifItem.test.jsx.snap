// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`first Debe de hacer match con el Snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="titulo"
      src="www.hola.com"
    />
    <p>
      titulo
    </p>
  </div>
</div>
`;
